### PR TITLE
[incubator/vault] Fix setting serviceaccount label

### PR DESCRIPTION
**Pull request by jbialy**
#### What this PR does / why we need it:

When attempting to deploy Vault with the serviceaccount flag enabled, i.e. `serviceAccount.create=true`, helm is unable to interpolate `vault.chart` as a label because the function is not defined, returning a templating error: `template "vault.chart" not defined`

This PR fixes this by setting the label using a combination of the `.Chart.Name` and `.Chart.Version`. Thanks to https://github.com/helm/charts/pull/9429#pullrequestreview-231684380 for pointing this out!

#### Which issue this PR fixes
  - Related to PR https://github.com/helm/charts/pull/9429

#### Checklist
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

Signed-off-by: Janusz Bialy <jbialy@gmail.com>

### CODE REVIEW

File: incubator/vault/templates/serviceaccount.yaml [hunk 1]
```diff
@@ -5,7 +5,7 @@ metadata:
   name: {{ template "vault.serviceAccountName" . }}
   labels:
     app: {{ template "vault.name" . }}
-    chart: {{ template "vault.chart" . }}
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}"
```
Comment (mattfarina) [inline incubator/vault/templates/serviceaccount.yaml]: This does not capture the full scope of what the `chart` function does. Would you mind adding the standard `chart` function to the `_helpers.tpl` file instead. That function reads:

```
{{/*
Create chart name and version as used by the chart label.
*/}}
{{- define "wtf.chart" -}}
{{- printf "%s-%s" .Chart.Name .Chart.Version | replace "+" "_" | trunc 63 | trimSuffix "-" -}}
{{- end -}}
```

Comment (jbialy) [reply incubator/vault/templates/serviceaccount.yaml]: Good call, I can do that. I'll also adjust the other labels that currently reference `{{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}` to use the `vault.chart` function.

**Comment by mattfarina** [comment]
@jbialy The failed test was due to chart versions. A retest won't fix those.

**Comment by jbialy** [comment]
> due to chart versions. A retest won't fix those.

Ah! I couldn't tell this by simply looking at the short version of the build log... I think I see it now! 😄 

```
I0430 15:38:08.786] Linting chart 'vault => (version: "0.18.4", path: "incubator/vault")'
I0430 15:38:08.786] Checking chart 'vault => (version: "0.18.4", path: "incubator/vault")' for a version bump...
I0430 15:38:08.791] Old chart version: 0.18.4
I0430 15:38:08.792] New chart version: 0.18.4
I0430 15:38:08.792] Error linting and installing charts: Error processing charts
I0430 15:38:08.792] ------------------------------------------------------------------------------------------------------------------------
I0430 15:38:08.792]  âœ–ï¸Ž vault => (version: "0.18.4", path: "incubator/vault") > Chart version not ok. Needs a version bump!
I0430 15:38:08.792] ------------------------------------------------------------------------------------------------------------------------
```

**Pull request merged**